### ISSUE TITLE
Site: Overclocking Support

### ISSUE THREAD

**Issue by Maelstromeous** [Maelstromeous/satisfactory-factories#11]
Add ability to overclock factory product lines.

Calculations to power will be a fun one.

**Comment by AndrewThomasWill** [comment]
Just thinking this through from the UI side of things, I think we could tie this to Product item
![image](https://github.com/user-attachments/assets/6d5806b7-ab1b-4059-ab98-aa52bfdd49eb)
Where you could maybe right-click (or double-click) these machines and it brings up a modal for clock speed, and sloops

(This is a very rough model, but I think it gets the point across 😄)
![chrome_uBmqeMyRaC](https://github.com/user-attachments/assets/de0d9d21-8669-4b3c-81cc-8cac1425bb57)

**Comment by Maelstromeous** [comment]
Think doing in tabular form might be better, as then the user at a glance can see where they've added power shards, Somersloops etc and can easily go round their factory and check.

**Comment by samsmithnz** [comment]
Tracking where Somersloops is a missing feature in all tools - that is useful. I always forget where I put them...

**Comment by AndrewThomasWill** [comment]
> @AndrewThomasWill Maybe rather than double clicking, we make the chip slightly bigger and add a Somersloop icon to it. We have tons of horizontal space, although I do have my doubts about readability.
> 
> Alternatively perhaps we split the area under the product section in two:
> 
> ```
> [PRODUCT] || [RECIPE] || [QTY] || [BUTTONS]
> _____
> Requirements                | Buildings
> [ICON] Iron plate 200/min   | 2x constructor [SLOOP ICON]: [0]x | [OC ICON]: [100%]x[1]
> [ICON] Iron rod 200/min     | 2x assembler   [SLOOP ICON]: [0]x | [OC ICON]: [100%]x[1]
> ----------                  | POWER: 1000-1500MW
> ```
> 
> Hopefully that makes some sense?

Yeah, I think this is taking the same approach, just formatting the data a lot better 👍
I still want to ensure we are on the same page about accounting for overclocking/slooping for a **group** of machines, not each machine. That should be a good middle ground from where we are now. I've only seen one other tool account for overclocking/slooping and I believe they do it the same way

**Comment by samsmithnz** [comment]
> I still want to ensure we are on the same page about accounting for overclocking/slooping for a **group** of machines, not each machine. The user could create multiple 'Products' that would allow them to work around it, but we wouldn't be providing that level of detail on this interface.

Can you elaborate on why this would apply to a group of machines and not individual? I'm not following how this would work - especially if - for example, you have 7 sloops and 2 manufacturers... you can put 4 in one, but only 3 in the second...

**Comment by AndrewThomasWill** [comment]
Exactly, that's a good scenario to bring up. So let's use the example data that @Maelstromeous gave. 
Let's say we let the user enter the amount of Sloops, and the overclock percentage (maybe just whole numbers for this example)
```
[PRODUCT] || [RECIPE] || [QTY] || [BUTTONS]
_____
Requirements                | Buildings
[ICON] Iron plate 200/min   | 2x constructor [SLOOP ICON]: [1]x | [OC ICON]: [135%]x[1]
[ICON] Iron rod 200/min     | 2x assembler   [SLOOP ICON]: [2]x | [OC ICON]: [137%]x[1]
----------                  | POWER: 1000-1500MW
```

When they enter '1x' sloop, is that assuming that both of the constructors have 1x sloop?
Or there is only 1 Constructor that has 1 sloop?
The same goes for Overclocking, is that number assuming 135% for both of the constructors? Or only 1?
(Reviewing back, I now see the x[1] for the percentage, so we could allow them to enter how many machine for overclocking - but currently not slooping)

If we want to let them specify, each individual machine in the group, with the number of sloops and the overclock percentage. I think we need a different approach.
(I've also not seen a Satisfactory tool let any get that granular before, but I could be wrong)
They could use the workaround I mentioned, where they duplicate the 'Product' and set different values:

```
[PRODUCT] || [RECIPE] || [QTY] || [BUTTONS]
_____
Requirements                | Buildings
[ICON] Iron plate 100/min   | 2x constructor [SLOOP ICON]: [1]x | [OC ICON]: [135%]x[1]
[ICON] Iron plate 100/min   | 2x constructor [SLOOP ICON]: [2]x | [OC ICON]: [145%]x[1]
[ICON] Iron rod 200/min     | 2x assembler   [SLOOP ICON]: [2]x | [OC ICON]: [137%]x[1]
----------                  | POWER: 1000-1500MW
```

If we instead want to assume those numbers apply to all machines **within** that group, that makes it a lot easier, but maybe isn't as granular config for the user.
I'm for the second option and only allowing it for the group. As I said before, I think it's a good middle ground from where we are now too. But I'm open to discussing this more to figure out the best solution 👍

**Comment by Maelstromeous** [comment]
My intention was while it is displayed as a group the sloops and overclock are individual machines. 

Slopping is fairly simple as you can only inject X sloops in a machine. Overclocking however you can have multiple machines with different clock settings so that's not as easy.

**Comment by Maelstromeous** [comment]
# Logical groups

This example has Wet Concrete at 50 refineries.

1. Create a tray that expands on a per product level
2. Add a + button to break up the 50 refineries into sub-groups
3. For each group, add the ability to add:
  1. Number of refineries
  2. Clock speed
  3. # sloops

This should give us the data required to run the math and produce the output of the item quantity.

**Comment by Morcis** [comment]
No reason to pay attention to me at all, but as a user with 90 current factories, allowing logical groups of like machines makes a lot of sense. Many times a factory includes 89 normal machines and 1 underclocked machine to allow for complete efficiency. Without groups I don't know how to do that without duplicating an internal product. 

Alternately you could go all in with the duplicated product line approach, give it a real ui and make it feel like a feature rather than a workaround.
